akka-stream/src/main/scala/akka/stream/IOResult.scala

```diff
@@ -56,6 +66,16 @@ object IOResult {
  * This exception signals that a stream has been completed by an onError signal
  * while there was still IO operations in progress.
  */
+@deprecated("use IOOperationIncompleteException", "2.6.0")
 final case class AbruptIOTerminationException(ioResult: IOResult, cause: Throwable)
     extends RuntimeException("Stream terminated without completing IO operation.", cause)
     with NoStackTrace
+
+/**
+ * This exception signals that a stream has been completed or has an error while
+ * there was still IO operations in progress
+ *
+ * @param count The number of bytes read/written up until the error
+ * @param cause cause
+ */
+final class IOOperationIncompleteException(val count: Long, cause: Throwable) extends RuntimeException(cause)
```

Comment:
should it include a message to RuntimeException for better toString and logging?